add: ability to not generate JVM 1.8+ annotation targets (KT-54550) (#3317)
JetBrains/kotlin-web-site: docs/topics/annotations.md

}
```

### Ability to not generate JVM 1.8+ annotation targets

> If a Kotlin annotation with the target `TYPE` is compiled with the JVM target 1.8+, this annotation gets 

Reviewer: ```suggestion
If a Kotlin annotation has `TYPE` among its Kotlin targets, this annotation gets 
```

`java.lang.annotation.ElementType.TYPE_USE` in its list of Java annotation targets. 
This is an issue for Android clients with API levels less than 26, which didn't have `TYPE_USE` in the API. 

> To compile against JDK 1.8 and avoid generating annotation targets `TYPE_USE` or `TYPE_PARAMETER`, use the new compiler argument `-Xno-new-java-annotation-targets`.

Reviewer: ```suggestion
To avoid generating annotation targets `TYPE_USE` or `TYPE_PARAMETER`, use the new compiler argument `-Xno-new-java-annotation-targets`.
```

}
```

### Ability to not generate JVM 1.8+ annotation targets

If a Kotlin annotation has the `TYPE` target among its Kotlin targets, this annotation gets 
`java.lang.annotation.ElementType.TYPE_USE` in its list of Java annotation targets. 

> The same way the `TYPE_PARAMETER` Kotlin’s target maps on the `java.lang.annotation.ElementType.TYPE_PARAMETER` Java’s target. 

Reviewer: ```suggestion
The same way the `TYPE_PARAMETER` Kotlin’s target maps to the `java.lang.annotation.ElementType.TYPE_PARAMETER` Java’s target. 
```

This is an issue for Android clients with API levels less than 26, which didn't have these targets in the API. 

> To avoid generating the TYPE_USE` and `TYPE_PARAMETER` annotation targets, use the new compiler argument `-Xno-new-java-annotation-targets`.

Reviewer 2: ```suggestion
To avoid generating the `TYPE_USE` and `TYPE_PARAMETER` annotation targets, use the new compiler argument `-Xno-new-java-annotation-targets`.
```

Author: Oh! Thanks!!!

> To avoid generating the TYPE_USE` and the `TYPE_PARAMETER` annotation targets, use the new compiler argument `-Xno-new-java-annotation-targets`.

Reviewer: ```suggestion
To avoid generating the TYPE_USE` and `TYPE_PARAMETER` annotation targets, use the new compiler argument `-Xno-new-java-annotation-targets`.
```
